pkg/server/batch_check.go

```diff
@@ -0,0 +1,148 @@
+package server
+
+import (
+	"context"
+	"errors"
+
+	serverErrors "github.com/openfga/openfga/pkg/server/errors"
+
+	"github.com/openfga/openfga/pkg/server/commands"
+
+	grpc_ctxtags "github.com/grpc-ecosystem/go-grpc-middleware/tags"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/openfga/openfga/pkg/middleware/validator"
+
+	"github.com/openfga/openfga/internal/condition"
+	"github.com/openfga/openfga/internal/graph"
+
+	openfgav1 "github.com/openfga/api/proto/openfga/v1"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/trace"
+
+	"github.com/openfga/openfga/internal/authz"
+	"github.com/openfga/openfga/pkg/telemetry"
+)
+
+func (s *Server) BatchCheck(ctx context.Context, req *openfgav1.BatchCheckRequest) (*openfgav1.BatchCheckResponse, error) {
+	ctx, span := tracer.Start(ctx, authz.BatchCheck, trace.WithAttributes(
+		attribute.KeyValue{Key: "store_id", Value: attribute.StringValue(req.GetStoreId())},
+		attribute.KeyValue{Key: "batch_size", Value: attribute.IntValue(len(req.GetChecks()))},
+		attribute.KeyValue{Key: "consistency", Value: attribute.StringValue(req.GetConsistency().String())},
+	))
+	defer span.End()
+
+	if !validator.RequestIsValidatedFromContext(ctx) {
+		if err := req.Validate(); err != nil {
+			return nil, status.Error(codes.InvalidArgument, err.Error())
+		}
+	}
+
+	err := s.checkAuthz(ctx, req.GetStoreId(), authz.BatchCheck)
+	if err != nil {
+		return nil, err
+	}
+
+	ctx = telemetry.ContextWithRPCInfo(ctx, telemetry.RPCInfo{
+		Service: s.serviceName,
+		Method:  authz.BatchCheck,
+	})
+
+	storeID := req.GetStoreId()
+
+	typesys, err := s.resolveTypesystem(ctx, storeID, req.GetAuthorizationModelId())
+	if err != nil {
+		return nil, err
+	}
+
+	cmd := commands.NewBatchCheckCommand(
+		s.checkDatastore,
+		s.checkResolver,
+		typesys,
+		commands.WithBatchCheckCommandCacheController(s.cacheController),
+		commands.WithBatchCheckCommandLogger(s.logger),
+		commands.WithBatchCheckMaxChecksPerBatch(s.maxChecksPerBatchCheck),
+		commands.WithBatchCheckMaxConcurrentChecks(s.maxConcurrentChecksPerBatch),
+	)
+
+	result, metadata, err := cmd.Execute(ctx, &commands.BatchCheckCommandParams{
+		AuthorizationModelID: typesys.GetAuthorizationModelID(),
+		Checks:               req.GetChecks(),
+		Consistency:          req.GetConsistency(),
+		StoreID:              storeID,
+	})
+
+	if err != nil {
+		telemetry.TraceError(span, err)
+		var batchValidationError *commands.BatchCheckValidationError
+		if errors.As(err, &batchValidationError) {
```

Comment:
Should also get this line covered.

Reply:
[covered just now](https://app.codecov.io/github/openfga/openfga/commit/d2fdf108d400336fd18d7f3a792cc7caa9c55981/indirect-changes#36a0baed52e93860bd7168f6656dfaaf-L36)